python/ambassador/envoy/v3/v3listener.py

```diff
@@ -721,13 +743,15 @@ def compute_tcpchains(self) -> None:
             if not group_host:  # cleartext
                 # Special case. No Host in a TCPMapping means an unconditional forward,
                 # so just add this immediately as a "*" chain.
-                self.add_chain("tcp", None, "*").add_tcphost(irgroup)
+                self.add_chain("tcp", None, "*", "*").add_tcphost(irgroup)
             else:  # TLS/SNI
                 context = tlscontext_for_tcpmapping(irgroup, self.config)
                 if not context:
                     irgroup.post_error("No matching TLSContext found, disabling!")
                     continue
-                self.add_chain("tcp", context, group_host).add_tcphost(irgroup)
+
+                sni = group_host.rsplit(":", 1)[0] if group_host else "*"
+                self.add_chain("tcp", context, group_host, group_host).add_tcphost(irgroup)
```

Comment:
yup good catch

Reply:
I'm actually going to push a fixup commit for this. I had originally done this for symmetry between HTTP and TCP but the TCPMapping already has `host` and `port`. The port is required but the host is optional if the user wants to do SNI and match against TLSContext. Therefore, for this PR I think it is out-of-scope and we can leave it. But, maybe it would be good to provide a earlier TCPMapping validation on the host to ensure the user hasn't done it. That to me though is a new feature and could be done in a follow-up PR.